### PR TITLE
First proposal for a simple feasibility process

### PR DESCRIPTION
**Pull request by wetret**

### CODE REVIEW

File: dsf-bpe/dsf-bpe-process-base/src/main/java/org/highmed/dsf/fhir/task/AbstractTaskMessageSend.java [hunk 1]
```diff
@@ -93,57 +97,69 @@ protected void sendTask(String targetOrganizationIdentifierValue, String process
...
+		{
+			logger.trace("Using local webservice client");
+			client = getFhirWebserviceClientProvider().getLocalWebserviceClient();
+		}
+		else
+		{
+			logger.trace("Using remote webservice client");
+			client = getFhirWebserviceClientProvider().getRemoteWebserviceClient(organizationProvider.getDefaultSystem(),
+					targetOrganizationIdentifierValue);
+		}
```
Comment (hhund) [inline dsf-bpe/dsf-bpe-process-base/src/main/java/org/highmed/dsf/fhir/task/AbstractTaskMessageSend.java]: refactor into a getClient(task) method

File: dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/CallActivityListener.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.highmed.dsf.bpe.listener;
+
+import org.camunda.bpm.engine.delegate.DelegateExecution;
+import org.camunda.bpm.engine.delegate.ExecutionListener;
+import org.highmed.dsf.bpe.Constants;
+
+public class CallActivityListener implements ExecutionListener
```
Comment (hhund) [inline dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/CallActivityListener.java]: Consider adding a few lines of documentation, explaining the purpose of this class within the framework.

File: dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/CallActivityListener.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+{
+	@Override
+	public void notify(DelegateExecution execution) throws Exception
+	{
+		execution.setVariable(Constants.VARIABLE_IS_CALL_ACTIVITY, true);
```
Comment (hhund) [inline dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/CallActivityListener.java]: please add a comment, why the VARIABLE_IS_CALL_ACTIVITY variable is set to true here.

File: dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/EndListener.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import org.camunda.bpm.engine.delegate.DelegateExecution;
+import org.camunda.bpm.engine.delegate.ExecutionListener;
+import org.highmed.dsf.bpe.Constants;
+import org.highmed.dsf.fhir.task.TaskHelper;
+import org.highmed.dsf.fhir.variables.OutputWrapper;
+import org.highmed.fhir.client.FhirWebserviceClient;
+import org.hl7.fhir.r4.model.Task;
+
+public class EndListener implements ExecutionListener
```
Comment (hhund) [inline dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/EndListener.java]: Consider adding a few lines of documentation, explaining the purpose of this class within the framework.

File: dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/StartListener.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.camunda.bpm.engine.delegate.DelegateExecution;
+import org.camunda.bpm.engine.delegate.ExecutionListener;
+import org.highmed.dsf.bpe.Constants;
+import org.highmed.dsf.fhir.variables.OutputWrapper;
+import org.hl7.fhir.r4.model.Task;
+
+public class StartListener implements ExecutionListener
```
Comment (hhund) [inline dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/StartListener.java]: Consider adding a few lines of documentation, explaining the purpose of this class within the framework.

File: dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/StartListener.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		if (execution.getParentId() == null || execution.getParentId().equals(execution.getProcessInstanceId()))
+		{
+			Task task = (Task) execution.getVariable(Constants.VARIABLE_TASK);
+			execution.setVariable(Constants.VARIABLE_LEADING_TASK, task);
+
+			execution.setVariable(Constants.VARIABLE_PROCESS_OUTPUTS, new ArrayList<OutputWrapper>());
+		}
+
+		if (!execution.getVariableNames().contains(Constants.VARIABLE_IS_CALL_ACTIVITY)) {
+			execution.setVariable(Constants.VARIABLE_IS_CALL_ACTIVITY, false);
```
Comment (hhund) [inline dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/StartListener.java]: consider adding a comment explaining lines 27 and 28

File: dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/variables/FinalSimpleFeasibilityResult.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package org.highmed.dsf.bpe.variables;
+
+import java.io.Serializable;
+
+public class FinalSimpleFeasibilityResult implements Serializable
```
[the remark this answers is not in the document]
Comment (wetret) [reply dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/variables/FinalSimpleFeasibilityResult.java]: this was already implemented once. I will search the previous commit and add it again.

File: dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/variables/MultiInstanceResult.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
+package org.highmed.dsf.bpe.variables;
+
+import java.io.Serializable;
+import java.util.Map;
+
+public class MultiInstanceResult implements Serializable
```
[the remark this answers is not in the document]
Comment (wetret) [reply dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/variables/MultiInstanceResult.java]: this was already implemented once. I will search the previous commit and add it again.

File: dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/variables/MultiInstanceResults.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
+package org.highmed.dsf.bpe.variables;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MultiInstanceResults implements Serializable
```
[the remark this answers is not in the document]
Comment (wetret) [reply dsf-bpe/dsf-bpe-process-feasibility/src/main/java/org/highmed/dsf/bpe/variables/MultiInstanceResults.java]: this was already implemented once. I will search the previous commit and add it again.

File: dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/DefaultBpmnParseListener.java [hunk 10]
```diff
@@ -9,14 +9,18 @@
 import org.camunda.bpm.engine.impl.util.xml.Element;
 import org.springframework.beans.factory.InitializingBean;
 
+/**
+ * Adds listener classes to certain types of BPMN activities before execution of a process
```
Comment (hhund) [inline dsf-bpe/dsf-bpe-server/src/main/java/org/highmed/dsf/bpe/listener/DefaultBpmnParseListener.java]: be precise, which activities are modified?

**Pull request merged**